pyproject.toml

```diff
@@ -67,13 +67,16 @@ test = [
     "pytest-mock",
     "pytest-reportlog",
 ]
-docs =[
+docs = [
     "myst-parser",
     "nbsphinx",
     "pydata-sphinx-theme",
     "sphinx",
     "sphinx-autoapi",
 ]
+all = [
+    "ecoscope[analysis,mapping,plotting,test,docs]",
+]
```

Comment:
Does this work? Awesome if so! Just didn't know this type of "self" reference (i.e., `ecoscope[...`) was supported here.

Reply:
I guess I would've assumed that `all` would require duplicative re-stating of the contents of `analysis`, `mapping`, etc., e.g. 

https://github.com/fsspec/filesystem_spec/blob/3ff5fcaff6a4845cdd1985594120105fb35ceb51/pyproject.toml#L35-L53

Reply:
I would have assumed the same, but did a quick search to see if something as above was possible, and it turns out it is :sunglasses: 

Interestingly this seems to be a lesser documented 'feature' of the ecosystem
[related pip issue](https://www.github.com/pypa/pip/issues/11296)

Reply:
TIL 🤯  That's awesome!